src/ape/api/providers.py

```diff
@@ -268,69 +339,162 @@ class ProviderAPI:
 
     @abstractmethod
     def connect(self):
-        ...
+        """
+        Connect a contract to a provider.
+        """
 
     @abstractmethod
     def disconnect(self):
-        ...
+        """
+        Disconnect a contract from a provider.
+        """
 
     @abstractmethod
     def update_settings(self, new_settings: dict):
-        ...
+        """
+        Change the settings connection protocol.
+        May require a reconnect.
+
+        Agrs:
+            new_settings ([`dict`]): Value of new provider
+        """
 
     @property
     @abstractmethod
     def chain_id(self) -> int:
-        ...
+        """
+        Gives value of blockchain id.
+
+        Returns:
+            int: value of blockchain id."""
 
     @abstractmethod
     def get_balance(self, address: str) -> int:
-        ...
+        """
+        Gives values of address balance.
+
+        Args:
+            address (`str`): Address string.
+
+        Returns:
+            int: value of balance at the address.
+        """
 
     @abstractmethod
     def get_code(self, address: str) -> bytes:
-        ...
+        """
+        Get the bytes of the contract.
+        Must override address or else raise `error`.
+
+        Args:
+            address (`str`): value of address (Required)
+
+        Returns:
+        bytes (`int`): HexBytes of contract.
+        """
 
     @abstractmethod
     def get_nonce(self, address: str) -> int:
-        ...
+        """
+        Gets the number of transactions.
+
+        Agrs:
+            address (`str`): string of address.
+
+        Returns:
+            int: value of nonce.
+        """
 
     @abstractmethod
     def estimate_gas_cost(self, txn: TransactionAPI) -> int:
-        ...
+        """
+        Gets the calculated gas cost.
+        See :class:`~ape.api.accounts.AccountAPI.call` to see calculation.
+
+        Args:
+            txn (:class:`~ape.api.providers.TransactionAPI`) : transaction.
+
+        Returns:
+            int: value of estimated gas cost.
+        """
 
     @property
     @abstractmethod
     def gas_price(self) -> int:
-        ...
+        """
+        Value of gas.
+
+        Returns:
+            int
+        """
 
     @property
     def priority_fee(self) -> int:
+        """
+        Value of miner tip to incentivize them
+        to include your transaction in a block.
+
+        Returns:
+            int: Value of fee.
+        """
         raise NotImplementedError("priority_fee is not implemented by this provider")
 
     @property
     def base_fee(self) -> int:
+        """
+        Value of the fee determinted by the network.
```

Comment:
```suggestion
        Value of the fee determined by the network.
```

Lol! "determinted" might be a good portmanteau for something else

Reply:
oh hahahaha